apps/prairielearn/src/lib/question-variant.sql

```diff
@@ -0,0 +1,76 @@
+-- BLOCK select_instance_question_data
+SELECT
+  aq.question_id,
+  ai.group_id,
+  ai.user_id,
+  iq.assessment_instance_id,
+  a.course_instance_id,
+  iq.open AS instance_question_open,
+  ai.open AS assessment_instance_open
+FROM
+  instance_questions AS iq
+  JOIN assessment_questions AS aq ON (aq.id = iq.assessment_question_id)
+  JOIN assessment_instances AS ai ON (ai.id = iq.assessment_instance_id)
+  JOIN assessments AS a ON (a.id = ai.assessment_id)
+WHERE
+  iq.id = $instance_question_id;
+
+-- BLOCK next_variant_number
+SELECT
+  max(v.number) + 1
+FROM
+  variants AS v
+WHERE
+  v.instance_question_id = $instance_question_id;
+
+-- BLOCK insert_workspace
+INSERT INTO
+  workspaces DEFAULT
+VALUES
+RETURNING
+  id;
```

Comment:
https://github.com/sql-formatter-org/sql-formatter/issues/718

Reply:
I opened https://github.com/sql-formatter-org/sql-formatter/pull/720 to improve this.